QueryGenerator.php

```diff
@@ -380,29 +395,37 @@ private function assertCompleteQuery() {
       }, $minimumClauses);
       $requiredClauseStr = '{' . implode('}, {', $requiredClauseOptions) . '}';
       throw new MissingRequiredClauseException(
-       "Missing required clauses. One of $requiredClauseStr needed.");
+         "Missing required clauses. One of $requiredClauseStr needed."
+      );
    }
 
    /**
     * Return the list of primary query clauses.
+    *
+    * @return non-empty-list<string>
     */
-   private static function getPrimaryClauses() {
+   private static function getPrimaryClauses(): array {
       return array_keys(self::$possibleClauses);
    }
 
    /**
     * Return the primary clause in this QueryGenerator instance.
     * If multiple primary clauses have been set, all but the first set clause
     * will be ignored.
+    *
+    * @return string|false
```

Comment:
This isn't necessary. `false` is a literal type now (as of 8.0):
https://php.watch/versions/8.0/union-types#false